File: problems/2176-count-equal-and-divisible-pairs-in-an-array/solution.cpp
// Link: https://leetcode.cn/problems/count-equal-and-divisible-pairs-in-an-array/
// Level: Medium
// Date: 2025-04-17

class Solution {
    public:
        int countPairs(vector<int>& nums, int k) {
            int n = nums.size();
            int cnt = 0;
            for(int i = 0; i < n; i++){
                for(int j = i + 1; j < n; j++){
                    if((nums[i] == nums[j]) && ((i * j) % k == 0)){
                        cnt++;
                    }
                }
            }
            return cnt;
        }
};

File: problems/2176-count-equal-and-divisible-pairs-in-an-array/solution.py
# Link: https://leetcode.cn/problems/count-equal-and-divisible-pairs-in-an-array/
# Level: Medium
# Date: 2025-04-17

class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        n = len(nums)
        cnt = 0
        for i in range(n):
            for j in range(i + 1, n):
                if nums[i] == nums[j] and (i * j) % k == 0:
                    cnt += 1
        return cnt
